app/controllers/buildings_controller.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpContext } from "@adonisjs/core/http";
+
+import Building from "#models/building";
+import { paginationValidator } from "#validators/pagination";
+import { showValidator } from "#validators/show";
+
+export default class BuildingsController {
+  /**
+   * Display a list of resource
+   */
+  async index({ request }: HttpContext) {
+    const { page, limit } = await request.validateUsing(paginationValidator);
+    const baseQuery = Building.query().withScopes((scopes) => {
+      scopes.handleSearchQuery(
+        request.only([
+          "id",
+          "identifier",
+          "specialName",
+          "campusId",
+          "addressLine1",
+          "addressLine2",
+          "latitude",
+          "longitude",
+          "haveFood",
+          "createdAt",
+          "updatedAt",
+        ]),
+      );
```

Comment:
> Or maybe the scope could just take the whole request and extract what it needs by itself?

Upside for manual pointing columns is to exclude and include just the columns you need. I intentionally excluded iconType column here since it does not have supported filtering type. This behaviour could be achieved by filtering object columns that have this type defined in meta property. But than again we loose the flexibility to exclude some other odd columns.

> Is there a way to do this without hard-coding every possible field in the controller?
```suggestion
        request.only([
          ...Object.keys(
            Building.$columnsDefinitions.entries().filter(([_, column]) => {
              const columnDef: ColumnDef = column;
              return columnDef.meta !== undefined
                ? columnDef.meta.type !== undefined
                : false;
            }),
          ),
        ]),
```
Could be done like so, dunno if I should even try to exclude columns with type other than allowed in this scope. Is it better than hardcoding? Hardcoded is more readable for sure, But suggested approach could  be more managable since you can get caught relying on word that "yes there's type for sure"

Reply:
Looking at my suggestion i've got an idea that maybe it's actually a good approach to move this logic inside the scope as you said and leave an option to exclude some columns by user as a side parameters and only those would be hardcoded ?

Reply:
that's actually a good point, I didn't even notice that the iconType field was missing from that list.
I think that it's more likely that we'd want to exclude just a few fields rather than including just a few select ones.